# Práctica Calificada 1

**Tópicos de Finanzas Avanzadas (ECON-421, UPAO 2026-20)**

Miércoles 16 de setiembre de 2026. Cubre las semanas 1 a 3. Puntaje total: 20 puntos. Tiempo sugerido: 45 minutos.

## Reglas del juego

Esta es una evaluación con **libro abierto de verdad**: pueden usar internet, sus apuntes, el repositorio del curso, Python y cualquier IA (ChatGPT, Claude, Gemini, la que quieran). No es trampa, es la forma en que van a trabajar como profesionales.

A cambio, las reglas son dos:

1. **Las respuestas escritas van con sus palabras y aplicadas a su caso.** Una respuesta genérica pegada de una IA, correcta pero que podría ser de cualquier alumno y cualquier empresa, obtiene la mitad del puntaje del ítem. Lo que evalúo es que ustedes entiendan, no que su IA entienda.
2. **El trabajo es individual.** Pueden conversar con una IA, no entre ustedes.

## Cómo se entrega

1. Completa este notebook (respuestas escritas en las celdas de texto, código donde se pida).
2. `Kernel  Restart & Run All` para verificar que corre completo.
3. Exporta a PDF: en Colab, menú Archivo, Imprimir, Guardar como PDF; en VS Code o Jupyter, Ctrl+P e imprimir a PDF.
4. Sube el PDF a Canvas antes del fin de la sesión. Como respaldo, haz también commit del notebook en tu fork.

## Parte 1: los conceptos con tus palabras (4 pts, 1 c/u)

Responde cada pregunta en un máximo de dos oraciones, **con tus palabras y usando tu empresa** (la de tus tareas 2 y 3) como ejemplo donde aplique.

**1.1** La semana 1 dijimos que el precio de mercado y el valor intrínseco de una acción pueden diferir. ¿Por qué esa diferencia no te dice, por sí sola, cuánto puedes ganar comprando la acción?

*Tu respuesta:* En Buenaventura, que mi valor estimado sea mayor que el precio de mercado no significa que esa diferencia sea una ganancia segura, porque puede incluir tanto un verdadero mispricing como errores en mis propios supuestos. Además, no sé cuánto de esa diferencia corresponde realmente a cada componente.

**1.2** ¿Por qué la tasa de descuento de tu empresa no es un número que se busca en internet sino un número que se construye? Nombra dos insumos que tuviste que decidir tú al construirla.

*Tu respuesta:* La tasa de descuento de Buenaventura no se encuentra como un dato único en internet, sino que se construye a partir de varios supuestos y decisiones del análisis. En mi caso, tuve que elegir la tasa libre de riesgo y la prima por riesgo de mercado que usaría para estimar el costo del equity.

**1.3** Explica a un gerente sin formación financiera por qué la utilidad neta de su empresa puede crecer mientras la caja disponible cae.

*Tu respuesta:* La utilidad neta de Buenaventura puede aumentar y, aun así, su caja disponible disminuir si la empresa destina más dinero a inversiones o al capital de trabajo. Por ejemplo, un mayor capex consume efectivo aunque ese desembolso no reduzca directamente la utilidad neta del mismo periodo.

**1.4** ¿Cuál es la diferencia entre lo que tu empresa podría repartir a sus accionistas y lo que efectivamente les reparte? ¿Cómo se llama cada concepto?

*Tu respuesta:* En Buenaventura, el FCFE representa la caja que podría quedar disponible para los accionistas después de cubrir las necesidades del negocio y la deuda. Los dividendos son lo que la empresa decide efectivamente repartir, por lo que no necesariamente son iguales al FCFE.

## Parte 2: encuentra los errores (6 pts)

Un practicante valoró a la empresa Andina SAC y entregó este trabajo. Sus datos de partida son correctos: EBIT = 50, t = 30 por ciento, depreciación = 10, capex = 15, incremento del capital de trabajo = 5, gasto por intereses = 8, endeudamiento neto = 0, Ke = 12 por ciento, WACC = 9 por ciento.

> **Paso 1.** FCFF = EBIT(1-t) + Dep + Int - Capex - WCInv = 35 + 10 + 8 - 15 - 5 = **33**
>
> **Paso 2.** FCFE = FCFF - Int = 33 - 8 = **25**
>
> **Paso 3.** Como la empresa creció 7 por ciento en los últimos años, uso g = 7 por ciento a perpetuidad. Valor del equity = FCFE por (1+g) entre (WACC - g) = 25 por 1.07 entre 0.02 = **1,337.5**

El trabajo contiene **cuatro errores conceptuales**. Identifícalos (2 pts), explica en una oración por qué cada uno es un error (2 pts), y recalcula el valor del equity corrigiendo todo, con un g perpetuo de 3 por ciento (2 pts). Puedes usar la celda de código.

*Errores y explicaciones:*

1.En el FCFF desde EBIT se sumó el interés.Al calcular el FCFF desde EBIT no se debe sumar el interés, porque el EBIT ya representa el resultado antes de intereses.

2.Para pasar de FCFF a FCFE se restó el interés completo. Para obtener el FCFE debe restarse el interés después de impuestos, no el interés completo

3.Usar \(g=7\%\) a perpetuidad solo porque la empresa creció 7% históricamente, es incorrecto.

4.El FCFE fue descontado al WACC. El FCFE corresponde únicamente al accionista, por lo que debe descontarse al costo del equity \(Ke\), no al WACC. La regla vista en la semana 2 es FCFF → WACC y FCFE → Ke.

In [1]:
# Recalculo corregido (puedes usar utils/finanzas.py si trabajas en tu fork)
EBIT = 50
t = 0.30
dep = 10
capex = 15
wc_inv = 5
interes = 8
endeudamiento_neto = 0
ke = 0.12
g = 0.03

fcff = EBIT * (1 - t) + dep - capex - wc_inv

fcfe = fcff - interes * (1 - t) + endeudamiento_neto

equity = fcfe * (1 + g) / (ke - g)

print("FCFF:", fcff)
print("FCFE:", fcfe)
print("Valor del equity:", equity)
# TU CODIGO AQUI

FCFF: 25.0
FCFE: 19.4
Valor del equity: 222.0222222222222


*Valor del equity corregido:* 222.0222222222222

## Parte 3: tu empresa (6 pts)

Con la empresa de tus tareas (usa `yfinance` o los resultados que ya tienes de la tarea 2):

**3.1** Calcula el FCFF del último año disponible por la ruta del EBIT. Muestra los insumos que usaste (EBIT, tasa efectiva, depreciación, capex, capital de trabajo). (3 pts)

In [2]:
import yfinance as yf
import pandas as pd

TICKER = "BVN"
empresa = yf.Ticker(TICKER)

income = empresa.income_stmt
cashflow = empresa.cashflow

# Buscar el último año con TODOS los datos necesarios
fechas_comunes = income.columns.intersection(cashflow.columns)

fecha = None

for f in sorted(fechas_comunes, reverse=True):

    datos_income = [
        income.loc["EBIT", f],
        income.loc["Tax Provision", f],
        income.loc["Pretax Income", f],
        income.loc["Reconciled Depreciation", f]
    ]

    datos_cashflow = [
        cashflow.loc["Capital Expenditure", f],
        cashflow.loc["Change In Working Capital", f]
    ]

    if all(pd.notna(x) for x in datos_income + datos_cashflow):
        fecha = f
        break

if fecha is None:
    raise ValueError("No se encontró un año con todos los datos necesarios.")

# Insumos
EBIT = float(income.loc["EBIT", fecha])

impuestos = float(income.loc["Tax Provision", fecha])
utilidad_antes_impuestos = float(income.loc["Pretax Income", fecha])

t = impuestos / utilidad_antes_impuestos

dep = float(income.loc["Reconciled Depreciation", fecha])

capex_raw = float(cashflow.loc["Capital Expenditure", fecha])
capex = abs(capex_raw)

cambio_wc = float(cashflow.loc["Change In Working Capital", fecha])

# Yahoo lo presenta como efecto sobre la caja
wc_inv = -cambio_wc

# FCFF por ruta EBIT
fcff = EBIT * (1 - t) + dep - capex - wc_inv

# Resultados
print("Año utilizado:", fecha.year)
print(f"EBIT: {EBIT/1e6:,.2f} millones")
print(f"Tasa efectiva: {t:.2%}")
print(f"Depreciación: {dep/1e6:,.2f} millones")
print(f"Capex: {capex/1e6:,.2f} millones")
print(f"Inversión en capital de trabajo: {wc_inv/1e6:,.2f} millones")
print("-" * 45)
print(f"FCFF: {fcff/1e6:,.2f} millones")

Año utilizado: 2023
EBIT: 180.54 millones
Tasa efectiva: 52.10%
Depreciación: 2.00 millones
Capex: 238.67 millones
Inversión en capital de trabajo: -4.95 millones
---------------------------------------------
FCFF: -145.24 millones


Para Buenaventura utilicé el año 2023, ya que fue el último periodo con información completa disponible en yfinance. Con un EBIT de 180.54 millones, una tasa efectiva de 52.10 %, depreciación de 2.00 millones, capex de 238.67 millones y una inversión en capital de trabajo de -4.95 millones, obtuve un FCFF de -145.24 millones. El FCFF negativo se explica principalmente por el elevado capex de 238.67 millones, que superó ampliamente la caja operativa generada después de impuestos.

In [3]:
# 3.2 Analizar deuda y FCFE de Buenaventura

balance = empresa.balance_sheet

# Trayectoria de la deuda
if "Total Debt" in balance.index:
    deuda = balance.loc["Total Debt"].dropna().sort_index()

    print("Trayectoria de la deuda:")
    for f, valor in deuda.items():
        print(f"{f.year}: {valor/1e6:,.2f} millones")
else:
    print("No se encontró la fila Total Debt")

# Intereses del año usado
interes = float(income.loc["Interest Expense", fecha])

# Endeudamiento neto
if "Net Issuance Payments Of Debt" in cashflow.index:
    endeudamiento_neto = cashflow.loc[
        "Net Issuance Payments Of Debt", fecha
    ]
else:
    endeudamiento_neto = (
        cashflow.loc["Issuance Of Debt", fecha]
        + cashflow.loc["Repayment Of Debt", fecha]
    )

endeudamiento_neto = float(endeudamiento_neto)

# FCFE
fcfe = fcff - interes * (1 - t) + endeudamiento_neto

print("\nAño analizado:", fecha.year)
print(f"FCFF: {fcff/1e6:,.2f} millones")
print(f"Intereses: {interes/1e6:,.2f} millones")
print(f"Endeudamiento neto: {endeudamiento_neto/1e6:,.2f} millones")
print(f"FCFE: {fcfe/1e6:,.2f} millones")

Trayectoria de la deuda:
2021: 50.00 millones
2023: 706.58 millones
2024: 9.17 millones
2025: 8.93 millones

Año analizado: 2023
FCFF: -145.24 millones
Intereses: 98.01 millones
Endeudamiento neto: -4.47 millones
FCFE: -196.67 millones


**3.2** Para valorar el equity de tu empresa, ¿usarías FCFF o FCFE? Justifica en dos o tres oraciones mirando su situación real: cómo ha venido cambiando su deuda y si su FCFE es positivo. (2 pts)

*Tu respuesta:* En Buenaventura usaría el FCFF porque su estructura de financiamiento ha tenido cambios relevantes y, además, para 2023 obtuve un FCFE negativo de -196.67 millones. La empresa mantenía obligaciones financieras importantes, principalmente por los bonos senior emitidos en 2021, por lo que el FCFF resulta más adecuado para valorar primero la firma y luego obtener el valor del equity.

**3.3** El interés que pagó tu empresa el año pasado, ¿a qué financista fue y por qué NO se resta al calcular el FCFF? (1 pt)

*Tu respuesta:* En Buenaventura, los 98.01 millones de intereses correspondieron al pago a los acreedores que financiaron a la empresa con deuda. No los resto en el FCFF porque este flujo se calcula antes de distribuir caja entre accionistas y acreedores; restarlos lo acercaría a un flujo exclusivo del accionista.

## Parte 4: tú y tu IA (4 pts)

Esta parte evalúa cómo usaste las herramientas, que es una competencia del curso.

**4.1** Pega el prompt más útil que le diste a una IA durante esta evaluación (o durante las tareas, si hoy no la usaste). (1 pt)

*Prompt:* Ayúdame a calcular el FCFF de Buenaventura (BVN) por la ruta del EBIT usando los datos de yfinance. Guíate por la fórmula de los PPT del curso, muéstrame cómo obtener EBIT, tasa efectiva, depreciación, capex y capital de trabajo, y dame el código en Python.

**4.2** Señala algo que la IA respondió de forma incompleta, genérica o directamente mal, y cómo lo detectaste o corregiste tú. Si crees que todo lo que te dijo estuvo perfecto, explica cómo lo verificaste. (3 pts)

*Tu respuesta:* Al inicio, la IA buscó la depreciación de Buenaventura únicamente dentro del estado de flujo de efectivo de yfinance, pero el código generó un KeyError porque esa fila no estaba disponible allí. Detecté el problema al revisar manualmente los nombres de las filas de cashflow e income_stmt; encontré Reconciled Depreciation en el estado de resultados y corregí el código para usar ese dato.

---

**Antes de entregar:** Restart & Run All, exportar a PDF, subir a Canvas. Commit en el fork como respaldo:

```bash
git add examenes/clase03_pc1.ipynb
git commit -m "PC1"
git push
```